File: src/main/java/com/wallet/app/service/AuthService.java
package com.wallet.app.service;

import com.wallet.app.dto.AuthResponse;
import com.wallet.app.dto.LoginRequest;
import com.wallet.app.dto.LoginResponse;
import com.wallet.app.dto.MessageResponse;
import com.wallet.app.dto.ResendEmailVerificationRequest;
import com.wallet.app.dto.RegisterRequest;
import com.wallet.app.dto.VerifyEmailRequest;
import com.wallet.app.dto.VerifyOtpRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse verifyEmail(VerifyEmailRequest request);

    MessageResponse resendEmailVerificationOtp(ResendEmailVerificationRequest request);

    LoginResponse login(LoginRequest request);

    LoginResponse verifyLoginOtp(VerifyOtpRequest request);
}

File: frontend/src/services/auth.service.ts
import type {
  AuthResponse,
  LoginRequest,
  LoginResponse,
  MessageResponse,
  RegisterRequest,
  ResendEmailVerificationRequest,
  VerifyEmailRequest,
  VerifyOtpRequest,
} from '../types/api'
import { apiClient } from './apiClient'

export const authService = {
  async register(payload: RegisterRequest): Promise<AuthResponse> {
    const response = await apiClient.post<AuthResponse>('/auth/register', payload)
    return response.data
  },

  async login(payload: LoginRequest): Promise<LoginResponse> {
    const response = await apiClient.post<LoginResponse>('/auth/login', payload)
    return response.data
  },

  async verifyEmail(payload: VerifyEmailRequest): Promise<AuthResponse> {
    const response = await apiClient.post<AuthResponse>('/auth/verify-email', payload)
    return response.data
  },

  async verifyLoginOtp(payload: VerifyOtpRequest): Promise<LoginResponse> {
    const response = await apiClient.post<LoginResponse>('/auth/verify-otp', payload)
    return response.data
  },

  async resendEmailVerificationOtp(payload: ResendEmailVerificationRequest): Promise<MessageResponse> {
    const response = await apiClient.post<MessageResponse>('/auth/resend-verification-otp', payload)
    return response.data
  },
}
